src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs

```diff
@@ -21,91 +28,66 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
 		// Get All Classes in User Library
 		var userGeneratedClassesProvider = context.SyntaxProvider.CreateSyntaxProvider(
 			static (syntaxNode, cancellationToken) => syntaxNode is ClassDeclarationSyntax { BaseList: not null },
-			static (context, cancellationToken) => (ClassDeclarationSyntax)context.Node);
-
-		// Get Microsoft.Maui.Controls Assembly Symbol
-		var mauiControlsAssemblySymbolProvider = context.CompilationProvider.Select(
-			static (compilation, token) => compilation.SourceModule.ReferencedAssemblySymbols.Single(q => q.Name == mauiControlsAssembly));
-
-		var inputs = userGeneratedClassesProvider.Collect()
-						.Combine(mauiControlsAssemblySymbolProvider)
-						.Select(static (combined, cancellationToken) => (UserGeneratedClassesProvider: combined.Left, MauiControlsAssemblySymbolProvider: combined.Right))
-						.Combine(context.CompilationProvider)
-						.Select(static (combined, cancellationToken) => (combined.Left.UserGeneratedClassesProvider, combined.Left.MauiControlsAssemblySymbolProvider, Compilation: combined.Right));
+			static (context, cancellationToken) =>
+			{
+				var compilation = context.SemanticModel.Compilation;
 
-		context.RegisterSourceOutput(inputs, (context, collectedValues) =>
-		Execute(context, collectedValues.Compilation, collectedValues.UserGeneratedClassesProvider, collectedValues.MauiControlsAssemblySymbolProvider));
-	}
+				var iTextStyleInterfaceSymbol = compilation.GetTypeByMetadataName(iTextStyleInterface);
+				var iAnimatableInterfaceSymbol = compilation.GetTypeByMetadataName(iAnimatableInterface);
 
-	static void Execute(SourceProductionContext context, Compilation compilation, ImmutableArray<ClassDeclarationSyntax> userGeneratedClassesProvider, IAssemblySymbol mauiControlsAssemblySymbolProvider)
-	{
-		var textStyleSymbol = compilation.GetTypeByMetadataName(iTextStyleInterface);
-		var iAnimatableSymbol = compilation.GetTypeByMetadataName(iAnimatableInterface);
-
-		if (textStyleSymbol is null || iAnimatableSymbol is null)
-		{
-			var diag = Diagnostic.Create(TextColorToDiagnostic.MauiReferenceIsMissing, Location.None);
-			context.ReportDiagnostic(diag);
-			return;
-		}
+				if (iTextStyleInterfaceSymbol is null || iAnimatableInterfaceSymbol is null)
+				{
+					throw new Exception("There's no .NET MAUI referenced in the project.");
```

Comment:
I don't think throwing an exception is correct if it's not caught and transformed to a diagnostic.